src/main/java/com/minecolonies/coremod/compatibility/jei/JEIPlugin.java

```diff
@@ -0,0 +1,46 @@
+package com.minecolonies.coremod.compatibility.jei;
+
+import com.minecolonies.coremod.compatibility.jei.transer.PrivateCraftingTeachingTransferHandler;
+import mezz.jei.api.IJeiRuntime;
+import mezz.jei.api.IModPlugin;
+import mezz.jei.api.IModRegistry;
+import mezz.jei.api.ISubtypeRegistry;
+import mezz.jei.api.ingredients.IModIngredientRegistration;
+import mezz.jei.api.recipe.IRecipeCategoryRegistration;
+import mezz.jei.api.recipe.VanillaRecipeCategoryUid;
+import mezz.jei.api.recipe.transfer.IRecipeTransferRegistry;
+
+@mezz.jei.api.JEIPlugin
+public class JEIPlugin implements IModPlugin
+{
+    @Override
+    public void registerItemSubtypes(final ISubtypeRegistry iSubtypeRegistry)
+    {
+
+    }
+
+    @Override
+    public void registerIngredients(final IModIngredientRegistration iModIngredientRegistration)
+    {
+
+    }
+
+    @Override
+    public void registerCategories(final IRecipeCategoryRegistration iRecipeCategoryRegistration)
+    {
+
+    }
+
+    @Override
+    public void register(final IModRegistry iModRegistry)
+    {
+        final IRecipeTransferRegistry recipeTranferRegistry = iModRegistry.getRecipeTransferRegistry();
+        recipeTranferRegistry.addRecipeTransferHandler(new PrivateCraftingTeachingTransferHandler(iModRegistry.getJeiHelpers().recipeTransferHandlerHelper()), VanillaRecipeCategoryUid.CRAFTING);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 194 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)